### DISCUSSION TITLE
Is there a provision to deploy the applications to different namespaces as well?

### DISCUSSION THREAD

**Discussion by raviusit** [argoproj-labs/applicationset#243] [Q&A]
I have a use case where I would like to deploy the same application in multiple clusters (for example - staging, prod, sales) but each cluster will have different namespace for this particular application. 
For example the naming convention followed for this namespace goes like this <appNameFamily>-<envType>-<commonSuffix>
aytscbec (Staging)
aytpcbec (Prod)
aytpsbec (Sales)

**Comment by rumstead** [comment]
You can use a generator(s) to template out the namespace. An example can be found [here](https://github.com/argoproj-labs/applicationset/blob/master/examples/matrix/list-and-git.yaml).

**Reply by nicolas-vivot** [reply]
The given example is using the same path.basename for both the metadata name and the destination namespace.
That means, each of your application will be deployed in a dedicated namespace that has the name of the application.. not really a real example.

With the git generator (directory base), and a folder structure "namespace/appName/charts", how do you extract the "appName" part ?
